mlops/docker_taxi/src/mlops_pipeline.py

```diff
@@ -77,252 +71,101 @@ def docker_taxi_data_regression(pipeline_job_input, model_name, build_reference)
+        train_model.environment = self.environment_name
+        predict_result.environment = self.environment_name
+        score_data.environment = self.environment_name
+        register_model.environment = self.environment_name
+
+        gl_pipeline_components.append(prepare_data)
+        gl_pipeline_components.append(transform_data)
+        gl_pipeline_components.append(train_model)
+        gl_pipeline_components.append(predict_result)
+        gl_pipeline_components.append(score_data)
+        gl_pipeline_components.append(register_model)
+
+        pipeline_job = docker_taxi_data_regression(
+            Input(type="uri_folder", path=registered_data_asset.id),
+            self.model_name,
+            self.build_reference,
         )
 
-        print(f"The job {pipeline_job.name} has been submitted!")
-        if output_file is not None:
-            with open(output_file, "w") as out_file:
-                out_file.write(pipeline_job.name)
-
-        if wait_for_completion == "True":
-            total_wait_time = 3600
-            current_wait_time = 0
-            job_status = [
-                "NotStarted",
-                "Queued",
-                "Starting",
-                "Preparing",
-                "Running",
-                "Finalizing",
-                "Provisioning",
-                "CancelRequested",
-                "Failed",
-                "Canceled",
-                "NotResponding",
-            ]
-
-            while pipeline_job.status in job_status:
-                if current_wait_time <= total_wait_time:
-                    time.sleep(20)
-                    pipeline_job = client.jobs.get(pipeline_job.name)
-
-                    print("Job Status:", pipeline_job.status)
-
-                    current_wait_time = current_wait_time + 15
-
-                    if (
-                        pipeline_job.status == "Failed"
-                        or pipeline_job.status == "NotResponding"
-                        or pipeline_job.status == "CancelRequested"
-                        or pipeline_job.status == "Canceled"
-                    ):
-                        break
-                else:
-                    break
-
-            if pipeline_job.status == "Completed" or pipeline_job.status == "Finished":
-                print("job completed")
-            else:
-                raise Exception("Sorry, exiting job with failure..")
-    except Exception as ex:
-        print(
-            "Oops! invalid credentials or error while creating ML environment.. Try again...",
-            ex,
-        )
-        raise
+        # demo how to change pipeline output settings
+        pipeline_job.outputs.pipeline_job_prepped_data.mode = "rw_mount"
+
+        return pipeline_job
 
 
 def prepare_and_execute(
-    build_environment: str,
-    wait_for_completion: str,
-    output_file: str,
+    model_name, build_environment, wait_for_completion, output_file
```

Comment:
let's add the variable type as part of the method signature

Reply:
updated signature